### ISSUE TITLE
BUG : Backfill -m is actually running tasks on LocalExecutor instead of just marking success

### ISSUE THREAD

**Issue by r39132** [apache/airflow#571]
`-m` should mark tasks as success without running them. Unfortunately, they are actually being run by the LocalExecutor! I'm running v 1.5.1

**Comment by r39132** [comment]
https://github.com/airbnb/airflow/issues/783

**Comment by r39132** [comment]
Closed thanks to #943 

Running a backfill with '-m' over multiple dates show this is working : 

```
id-As-MacBook-Pro:foo siddharth$ tail -f *

==> 2016-02-01T00:00:00 <==
[2016-02-07 21:32:49,459] {models.py:295} INFO - Loaded DAG <DAG: sid_test_5>
[2016-02-07 21:32:50,488] {models.py:126} INFO - Filling up the DagBag from /Users/siddharth/Projects/airflow/dags/sid_test_5.py
[2016-02-07 21:32:50,488] {models.py:204} INFO - Importing /Users/siddharth/Projects/airflow/dags/sid_test_5.py
[2016-02-07 21:32:50,488] {models.py:295} INFO - Loaded DAG <DAG: sid_test_5>
[2016-02-07 21:32:50,503] {models.py:988} INFO - 
--------------------------------------------------------------------------------
Attempt 1 out of 4
--------------------------------------------------------------------------------

[2016-02-07 21:32:50,510] {models.py:1028} INFO - Marking success for <Task(PythonOperator): foo> on 2016-02-01 00:00:00

==> 2016-02-01T01:00:00 <==
[2016-02-07 21:33:00,850] {models.py:295} INFO - Loaded DAG <DAG: sid_test_5>
[2016-02-07 21:33:01,812] {models.py:126} INFO - Filling up the DagBag from /Users/siddharth/Projects/airflow/dags/sid_test_5.py
[2016-02-07 21:33:01,813] {models.py:204} INFO - Importing /Users/siddharth/Projects/airflow/dags/sid_test_5.py
[2016-02-07 21:33:01,813] {models.py:295} INFO - Loaded DAG <DAG: sid_test_5>
[2016-02-07 21:33:01,830] {models.py:988} INFO - 
--------------------------------------------------------------------------------
Attempt 1 out of 4
--------------------------------------------------------------------------------

[2016-02-07 21:33:01,837] {models.py:1028} INFO - Marking success for <Task(PythonOperator): foo> on 2016-02-01 01:00:00

==> 2016-02-01T02:00:00 <==
[2016-02-07 21:33:12,215] {models.py:295} INFO - Loaded DAG <DAG: sid_test_5>
[2016-02-07 21:33:13,201] {models.py:126} INFO - Filling up the DagBag from /Users/siddharth/Projects/airflow/dags/sid_test_5.py
[2016-02-07 21:33:13,201] {models.py:204} INFO - Importing /Users/siddharth/Projects/airflow/dags/sid_test_5.py
[2016-02-07 21:33:13,201] {models.py:295} INFO - Loaded DAG <DAG: sid_test_5>
[2016-02-07 21:33:13,224] {models.py:988} INFO - 

```
